### PR TITLE
Mus 94 refactor piano view

### PR DESCRIPTION
**Pull request by AndrejMueller01**

### CODE REVIEW

**Comment by SimonStefan** [comment]
Your pull request isn't mergeable. Please fix it.
Further improvements:
- don't change the following files. Which program are you using for git? Have you set up your working environment correctly? [1]
  - .idea/encodings.xml
  - .idea/misc.xml
  - Musicdroid.iml
  - musicdroid/musicdroid.iml
  - musicdroidSourceTest/musicdroidSourceTest.iml
- don't put jar-Files into the repository if they are available on Maven Central!! [2]
- this pull request is far too big. And why are there e.g. break, cross and tact images in there? I think at the planning game these features where not planned.

[1] https://github.com/Catrobat/Musicdroid/wiki
[2] http://search.maven.org/#artifactdetails%7Ccom.jayway.android.robotium%7Crobotium%7C5.2.1%7Cpom

**Comment by bteufl** [comment]
Why is the very first pull request not mergeable?

File: musicdroid/src/main/java/org/catrobat/musicdroid/pocketmusic/instrument/piano/PianoViewFragment.java [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+        rootView.setLayoutParams(new ViewGroup.LayoutParams(ViewGroup.LayoutParams.MATCH_PARENT,getDisplayHeight()/2));
+
+        findViewsById(rootView);
+
+        calculatePianoKeyPositions(DEFAULT_PIANO_KEY_HEIGHT_SCALE_FACTOR,
+                DEFAULT_BLACK_KEY_WIDTH_SCALE_FACTOR);
+
+        setBlackKeyInvisible(Octave.DEFAULT_INACTIVE_BLACK_KEY);
+
+        // TODO: optimize touch events
```
Comment (bteufl) [inline musicdroid/src/main/java/org/catrobat/musicdroid/pocketmusic/instrument/piano/PianoViewFragment.java]: Why is there a todo?

**Comment by bteufl** [comment]
The break-images and the tact-image is not needed... In fact, no picture should be needed within the task refactor piano-view...

**Comment by SimonStefan** [comment]
You don't have to close a pull request if there are suggested improvements. You could just push the improvements to the same branch and the pull request will be updated automatically.

To clarify my comment: don't ever change the files I've mentioned

**Comment by AndrejMueller01** [comment]
Ok.

**Comment by fwinkelbauer** [comment]
https://jenkins.catrob.at/view/All-Categories/view/Musicdroid/job/Musicdroid-Custom-Branch-Gradle-Emulator/44/

File: .gitignore [hunk 2]
```diff
@@ -23,14 +23,11 @@ local.properties
 proguard/
 
 # IntelliJ config files
-/.idea/workspace.xml
```
Comment (SimonStefan) [inline .gitignore:26]: Please do not change the .gitignore! I played around at least 10 hours to find out the best solution for about 100 students using shared settings and different platforms. There is an instruction of the project setup in the Github Wiki you should have read in the first place. Now please do the following:
- undo the changes to .gitignore and commit it
- delete all unused branches! (there could be huge problems!)
- close Android Studio!
- execute in Terminal in project root (all written code should be committed, or will be lost)
  - "git reset --hard" should reset the iml-files and misc.xml
  - "git clean -xdf" should delete some hidden settings files
- proceed last 3 steps: https://github.com/Catrobat/Musicdroid/wiki

PS: in the current version of Android Studio Google fixed a bug. So maybe there will be an updated version of these settings without the iml-files soon.

**Pull request closed without merging**